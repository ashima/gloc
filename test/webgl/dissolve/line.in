// Copyright 2012 Foo <http://www.example.com/> Blah.
#line 2 0 // #n=module_a
#define DBL(t) \
t dbl(t a) { return a*2; }
DBL(float)
DBL(vec2)
DBL(vec3)
DBL(vec4)

// Copyright 2012 Bar <http://www.example.net/> Baz.
#line 9 1 // #n=module_b
#define AVG(t) \
t avg(t a, t b) { return (a+b)/2.0; }
AVG(float)
AVG(vec2)
AVG(vec3)
AVG(vec4)
