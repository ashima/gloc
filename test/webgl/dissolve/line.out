[
  [
    "[stdin]",
    {
      "glo": [ 1, 0, 0 ],
      "target": [ "webgl", [ 1, 0, 0 ] ],
      "meta": { "copyright": [ 2012, [ "Foo", "http://www.example.com/" ] ] },
      "units": [
        {
          "outmac": [ "DBL" ],
          "source":
            "#line 2 GLOC_0\n#define DBL(t)/*\\\n*/ t dbl( t a ) { return a * 2 ; }"
        },
        {
          "outsym": [ "dbl" ],
          "inmac": [ "DBL" ],
          "source":
            "#line 4 GLOC_0\nDBL(float)\nDBL(vec2)\nDBL(vec3)\nDBL(vec4)"
        }
      ],
      "linkmap": { "0": "#n=0" }
    }
  ],
  [
    "[stdin]",
    {
      "glo": [ 1, 0, 0 ],
      "target": [ "webgl", [ 1, 0, 0 ] ],
      "meta": { "copyright": [ 2012, [ "Bar", "http://www.example.net/" ] ] },
      "units": [
        {
          "outmac": [ "AVG" ],
          "source":
            "#line 9 GLOC_1\n#define AVG(t)/*\\\n*/ t avg( t a , t b ) { return ( a + b ) / 2.0 ; }"
        },
        {
          "outsym": [ "avg" ],
          "inmac": [ "AVG" ],
          "source":
            "#line 11 GLOC_1\nAVG(float)\nAVG(vec2)\nAVG(vec3)\nAVG(vec4)"
        }
      ],
      "linkmap": { "0": "#n=0", "1": "#n=module_b" }
    }
  ]
]
